src/lib/response-triggers/roleconfig.ts

```diff
@@ -0,0 +1,138 @@
+import Trigger from '../../models/trigger';
+import Config from '../config';
+import constants from '../../constants';
+import TableHelper from '../parsers/tableHelper';
+import { MessageObject } from 'ciscospark/env';
+import { Role, RolesRow } from '../../models/database';
+import { ConfigAction } from '../../models/config-action';
+
+export default class RoleConfig extends Trigger {
+	readonly commandText : string = 'roleconfig';
+	readonly roleConfigCommand : string = `(?: )*${this.commandText}(?: )*`;
+
+	config : Config;
+
+	constructor(config : Config) {
+		super();
+
+		this.config = config;
+	}
+
+	isToTriggerOn(message : MessageObject) : boolean {
+		if (!(this.config.checkRole(message.personId, Role.Admin) || this.config.checkRole(message.personId, Role.Config))) {
+			return false;
+		}
+		let pattern = new RegExp('^' + constants.optionalMarkdownOpening + constants.mentionMe + this.roleConfigCommand, 'ui');
+		return pattern.test(message.html);
+	}
+
+	isToTriggerOnPM(message : MessageObject) : boolean {
+		if (!(this.config.checkRole(message.personId, Role.Admin) || this.config.checkRole(message.personId, Role.Config))) {
+			return false;
+		}
+		return message.text.toLowerCase().trim().startsWith(this.commandText);
+	}
+
+	async createMessage(message : MessageObject) : Promise<MessageObject> {
+		message.text = message.text.toLowerCase();
+		message.text = message.text.trim();
+
+		let words = message.text.split(' ');
```

Comment:
Why not use the XML parser and grab the mentioned user, rather than require their userid be there?